dbms/src/Storages/DeltaMerge/StoragePool.cpp

```diff
@@ -187,21 +187,58 @@ StoragePool::StoragePool(NamespaceId ns_id_, const GlobalStoragePool & global_st
     , data_storage_reader(ns_id, data_storage, nullptr)
     , meta_storage_reader(ns_id, meta_storage, nullptr)
     , global_context(global_ctx)
+    , v3_log_max_ids(global_storage_pool.getLogMaxIds())
+    , v3_data_max_ids(global_storage_pool.getDataMaxIds())
+    , v3_meta_max_ids(global_storage_pool.getMetaMaxIds())
 {}
 
 void StoragePool::restore()
 {
     // If the storage instances is not global, we need to initialize it by ourselves and add a gc task.
     if (owned_storage)
     {
-        log_storage->restore();
-        data_storage->restore();
-        meta_storage->restore();
+        auto log_max_ids = log_storage->restore();
+        auto data_max_ids = data_storage->restore();
+        auto meta_max_ids = meta_storage->restore();
+
+        assert(log_max_ids.size() == 1);
```

Comment:
We don't use assert in normal logic besides test cases. Manually throw exceptions instead.

Reply:
That make no sense if `log_max_ids.size() != 1`. And I don't think using `exception` replace `assert` is a good thing.
IMO: 
1. If it's logically clear and you do know it inited where. Then used `assert` to check .
2. If some value changes a lot of time, Then the `exception` will replace the `assert` to avoid some problems.

So back to this `assert`, it's clear to know that `log_max_ids` from V2 must have `size 1`. So I think just adding an assert is ok to avoid logical error?

Reply:
I think assert is OK here